src/wp-includes/rest-api/endpoints/class-wp-rest-block-pattern-categories-controller.php

```diff
@@ -152,11 +192,45 @@ public function get_item_schema() {
 					'readonly'    => true,
 					'context'     => array( 'view', 'edit', 'embed' ),
 				),
+				'id'          => array(
+					'description' => __( 'An optional category id, currently used to provide id for user wp_pattern_category terms' ),
+					'type'        => 'number',
+					'readonly'    => true,
+					'context'     => array( 'view', 'edit', 'embed' ),
+				),
 			),
 		);
 
 		$this->schema = $schema;
 
 		return $this->add_additional_fields_schema( $this->schema );
 	}
+
+	/**
+	 * Retrieves the search parameters for the block pattern categories.
+	 *
+	 * @since 6.5.0 Added source param to request.
+	 *
+	 * @return array Collection parameters.
+	 */
+	public function get_collection_params() {
+		$query_params = parent::get_collection_params();
+
+		$query_params['source'] = array(
+			'description' => __( 'Limit result set to specific sources, `core` and/or `user`' ),
+			'type'        => 'array',
+			'items'       => array(
+				'type' => 'string',
```

Comment:
We can specify the allowed sources using the `enum` keyword.

Reply:
ok, will add that.